### PR TITLE
Implement `target_application` and rename `target_user_type` to `target_type`

### PR DESCRIPTION
**Pull request by davfsa**
### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: hikari/impl/entity_factory.py [hunk 1]
```diff
@@ -276,6 +277,14 @@ def deserialize_own_guild(self, payload: data_binding.JSONObject) -> application
             my_permissions=permission_models.Permissions(int(payload["permissions"])),
         )
 
+    # def deserialize_partial_application(self, payload: data_binding.JSONObject) -> guilds.PartialApplication:
```
[the remark this answers is not in the document]
Comment (davfsa) [reply hikari/impl/entity_factory.py:280]: This wasn't meant to see the light of day, just testing something out, xD

File: hikari/applications.py [hunk 2]
```diff
@@ -483,8 +483,8 @@ class Application(guilds.PartialApplication):
     Will be `builtins.None` if this application doesn't have a bot.
     """
 
-    owner: users.User = attr.field(eq=False, hash=False, repr=True)
-    """The application's owner."""
+    owner: typing.Optional[users.User] = attr.field(eq=False, hash=False, repr=True)
+    """The application's owner or `None` if owned by Discord."""
```
[the remark this answers is not in the document]
Comment (davfsa) [reply hikari/applications.py:487]: My PR on Discord API Docs was merged https://github.com/discord/discord-api-docs/commit/c95833903067782e12c36b598bf04b20e81d2932 regarding this. I will have to test some stuff out, but it seems it is the case.

[the remark this answers is not in the document]
Comment (FasterSpeeding) [reply hikari/applications.py:487]: The pr probably got merged cause it isn't included in a lot of places and the docs are like the worst at documenting partial cases. But user isn't included on partial applications in places you can only access with auth (including user only stuff funnily enough) so I don't see why it'd ever be included in the response on a public endpoint.

File: hikari/impl/entity_factory.py [hunk 3]
```diff
@@ -307,8 +316,8 @@ def deserialize_application(self, payload: data_binding.JSONObject) -> applicati
             description=payload["description"] or None,
             is_bot_public=payload.get("bot_public"),
             is_bot_code_grant_required=payload.get("bot_require_code_grant"),
-            owner=self.deserialize_user(payload["owner"]),
-            rpc_origins=payload["rpc_origins"] if "rpc_origins" in payload else None,
+            owner=self.deserialize_user(payload["owner"]) if "owner" in payload else None,
```
[the remark this answers is not in the document]
Comment (davfsa) [reply hikari/impl/entity_factory.py:322]: Moved out :+1:

File: hikari/api/entity_factory.py [hunk 4]
```diff
@@ -143,6 +143,21 @@ def deserialize_own_guild(self, payload: data_binding.JSONObject) -> application
             The deserialized "own guild" object.
         """
 
+    @abc.abstractmethod
+    def deserialize_invite_application(self, payload: data_binding.JSONObject) -> application_models.InviteApplication:
```
Comment (FasterSpeeding) [inline hikari/api/entity_factory.py:147]: Don't think we need a dedicated method for deserializing this since it's only ever found on invite objects

**Review by FasterSpeeding — APPROVED** [review]

**Pull request merged**